### PR TITLE
feat: support aten index_put converter for accumulate=False

### PR DESCRIPTION
**Pull request by chohk88**
# Description

I have implemented the `aten::index_put` operation using the `add_scatter` layer with `trt.ScatterMode.ND`. However, I was unable to implement the `accumulate=True` case, which is currently handled by the validator.

Fixes # ([issue](https://github.com/pytorch/TensorRT/issues/2544))

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/impl/select.py [hunk 1]
```diff
@@ -428,3 +429,41 @@ def scatter(
...
+    source_ir: Optional[SourceIR],
+    name: str,
+    input_tensor: TRTTensor,
+    indices: Union[TRTTensor, Tuple[TRTTensor, ...]],
+    values: TRTTensor,
+    accumulate: bool = False,
+) -> TRTTensor:
+    # Reshape indices to add an extra dimension if necessary (indices is a Tuple of ITensors)
+    reshaped_indices = []
+    for i, each_input in enumerate(indices):
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/conversion/impl/select.py:446]: Since `indices` is possible to be `ITensor` per the schema, you may not be able to iterate an ITensor.
In the test case, you can try to change the line 173 to `inputs=[source_tensor, indices_tensor, value_tensor],`.
It's kind of similar to the `offsets` in the annoying `embedding_bag`. You can think about how to use native TRT Layers to do this, like ILoop.

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/impl/select.py:446]: Besides, what blocks you when `accumulate=True`?

Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/impl/select.py:446]: When `accumulate=True`, if there are duplicate pairs of index in `indices`, the corresponding `values` should be summed and then removed from the elements. Therefore, I aimed to obtain `indices` without duplicated pairs and corresponding modified `values`, and then use these to input into the scatter layer. However, I encountered difficulties in implementing the for loop to check for duplicate pairs of index in `indices`.

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/impl/select.py:446]: Thanks for the detailed explanations! Yes you are right, the `indices` should be list or tuple, and thus it could be iterated over. Then your current implementation LGTM.
> One more question I have is about the type definition of indices when it is a tuple of tensors. Is it correct to define indices as Union[TRTTensor, Tuple[TRTTensor, ...]]?

I think it could be `Sequence[Union[TRTTensor, np.ndarray, torch.Tensor]]` since a single TRTTensor cannot be iterated, right?

File: tests/py/dynamo/conversion/test_index_put_aten.py
Comment (apbose) [inline tests/py/dynamo/conversion/test_index_put_aten.py:1]: The implementation looks good to me. Can you add a test case like this -
tensor = torch.zeros([4, 4, 4, 4], dtype = torch.int32)
indices = (torch.tensor([0,1,2,3]), torch.tensor([2,3, 1, 0]))
values = torch.tensor([10, 20, 30, 40],  dtype = torch.int32)
out = torch.index_put_(tensor, indices, values)

File: py/torch_tensorrt/dynamo/conversion/impl/select.py (hunk 1 above)
Comment (apbose) [reply py/torch_tensorrt/dynamo/conversion/impl/select.py:446]: When you say `accumulate=True` is causing issue, I believe the duplicate indices causes issues. I faced the same in `scatter_reduce` and I believe advanced indexing would be the way to deal with it (lengthy code that would be I believe :( ). Do you have any other ideas?

File: tests/py/dynamo/conversion/test_index_put_aten.py
Comment (narendasan) [reply tests/py/dynamo/conversion/test_index_put_aten.py:1]: Lets write a validator for this case and resolve in a new PR

Comment (chohk88) [reply tests/py/dynamo/conversion/test_index_put_aten.py:1]: I have write validator for broadcasting.

File: py/torch_tensorrt/dynamo/conversion/impl/select.py (hunk 1 above)
Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/impl/select.py:464]: I have write a validator to handle the `accumulate=True` case. And I have created a separate [issue](https://github.com/pytorch/TensorRT/issues/2939) for implementing the converter for `accumulate=True`. It would be great to share ideas and work together on this.
